File: BotsSDK/korebotsdklib/src/main/java/kore/botssdk/models/BrandingFooterButtonsModel.java
package kore.botssdk.models;

import java.io.Serializable;

@SuppressWarnings("UnKnownNullness")
public class BrandingFooterButtonsModel implements Serializable {
    private BrandingFooterMenuButtonModel menu;
    private BrandingIconModel emoji;
    private BrandingIconModel microphone;
    private BrandingIconModel attachment;
    private BrandingIconModel speaker;

    public BrandingFooterButtonsModel updateWith(BrandingFooterButtonsModel configModel) {
        menu = menu != null && configModel.menu != null ? menu.updateWith(configModel.menu) : menu;
        emoji = emoji != null && configModel.emoji != null ? emoji.updateWith(configModel.emoji) : emoji;
        microphone = microphone != null && configModel.microphone != null ? microphone.updateWith(configModel.microphone) : microphone;
        attachment = attachment != null && configModel.attachment != null ? attachment.updateWith(configModel.attachment) : attachment;
        speaker = speaker != null && configModel.speaker != null ? speaker.updateWith(configModel.speaker) : speaker;
        return this;
    }

    public BrandingFooterMenuButtonModel getMenu() {
        return menu;
    }

    public BrandingIconModel getAttachment() {
        return attachment;
    }

    public BrandingIconModel getSpeaker() {
        return speaker;
    }

    public BrandingIconModel getEmoji() {
        return emoji;
    }

    public BrandingIconModel getMicrophone() {
        return microphone;
    }
}

File: BotSdk/botclient/src/main/java/com/kore/network/api/responsemodels/branding/BrandingFooterButtonsModel.kt
package com.kore.network.api.responsemodels.branding

import com.google.gson.annotations.SerializedName

data class BrandingFooterButtonsModel(
    @SerializedName("menu")
    val menu: BrandingFooterMenuButtonModel,
    @SerializedName("emoji")
    val emoji: BrandingIconModel,
    @SerializedName("microphone")
    val microphone: BrandingIconModel,
    @SerializedName("attachment")
    val attachment: BrandingIconModel,
    @SerializedName("speaker")
    val speaker: BrandingIconModel
)

fun BrandingFooterButtonsModel.updateWith(configModel: BrandingFooterButtonsModel?): BrandingFooterButtonsModel {
    return this.copy(
        menu = this.menu.updateWith(configModel?.menu),
        emoji = this.emoji.updateWith(configModel?.emoji),
        microphone = this.microphone.updateWith(configModel?.microphone),
        attachment = this.attachment.updateWith(configModel?.attachment),
        speaker = this.speaker.updateWith(configModel?.speaker),
    )
}